api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java

```diff
@@ -429,7 +429,7 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
     }
 
     private boolean canContainNulls(Integer id) {
-      return nullCounts == null || (nullCounts.containsKey(id) && nullCounts.get(id) > 0);
+      return nullCounts == null || nullCounts.containsKey(id) && nullCounts.get(id) > 0;
```

Comment:
I think it is much better to have the parentheses so that the logic is clear to a reader, even if it is identical to Java's precedence. Can you update the style config and revert these changes?

Reply:
@rdblue pushed a new version where the `UnnecessaryParentheses` checkstyle module was removed